transformers/tyntecWhatsAppTransformer/transformer-v4.ts

```diff
@@ -0,0 +1,473 @@
+	let sessionquickReplyCurrentNumber: number = sessionStorage.quickReplyCurrentNumber || 0;
+	let sessionQuickReplies: ISessionStorageQuickReply[] = sessionStorage.quickReplies || [];
+
+	// initialize empty text message bubble
+	let whatsAppQuickReplyMessage: string = "";
+
+	for (let quickReply of quickReplies) {
+		// store the index to the session storage for further quick replies
+		sessionquickReplyCurrentNumber += 1;
+		sessionQuickReplies.push({
+			index: sessionquickReplyCurrentNumber,
+			quickReply
+		})
+		// add the quick reply to the text message bubble
+		// Example: 1. first quick reply
+		whatsAppQuickReplyMessage += `\n${sessionquickReplyCurrentNumber}. ${quickReply.title}`;
+
+	}
+
+	sessionStorage.quickReplyCurrentNumber = sessionquickReplyCurrentNumber;
+	sessionStorage.quickReplies = sessionQuickReplies;
+
+	return whatsAppQuickReplyMessage;
+}
+
+const convertWebchatContentToWhatsApp = (processedOutput, sessionId: string, sessionStorage: any): TWhatsAppContent[] => {
+
+	// create list for whatsapp content
+	let whatsAppContents: TWhatsAppContent[] = [];
+
+	// Loop through all provided Cogngiy bot messages
+	if (processedOutput.outputStack) {
+		for (let stackItem of processedOutput.outputStack) {
+
+			// check if default text was sent
+			if (stackItem.text && !(stackItem.data && stackItem.data?._cognigy && (stackItem.data?._cognigy?._default))) {
+
+				// send default text
+				whatsAppContents.push({
+					from: sessionId,
+					contentType: "text",
+					text: stackItem.text
+				});
+			}
+
+			// check for location message
+			else if (stackItem.data.location) {
+
+				const { longitude, latitude, name, address } = stackItem.data.location;
+
+				whatsAppContents.push({
+					from: sessionId,
+					contentType: "location",
+					location: {
+						longitude,
+						latitude,
+						name,
+						address
+					},
+				});
+			}
+
+			// check if default templates are defined
+			else if (stackItem.data && (stackItem.data?._cognigy?._default)) {
+
+				let defaultContent = stackItem.data._cognigy._default;
+
+				// look for gallery
+				if (defaultContent._gallery != null && defaultContent._gallery?.items) {
+					// look for galleries
+					const galleryElements = defaultContent._gallery.items;
+
+					// create gallery message as message bubble
+					for (let element of galleryElements) {
+
+						// check if image is provided
+						if (element.imageUrl === "" || element.imageUrl === null || element.imageUrl === undefined) {
+							throw new Error('[WhatsApp] Gallery item is missing image url');
+						}
+
+						whatsAppContents.push({
+							from: sessionId,
+							contentType: "media",
+							media: {
+								type: "image",
+								url: element.imageUrl,
+								caption: `*${element.title}*\n\n${element.subtitle}`
+							}
+						});
+
+						console.log(element)
+						// check for buttons and show them as quick replies
+						if (element.buttons && element.buttons?.length !== 0) {
```

Comment:
can be shortened to (element.buttons?.length)

Reply:
done